File: GO/TopN_test.go
/**
 * @version GO
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ go test
 */
package GO

import "testing"

func TestTopN(t *testing.T) {
	tables := []struct {
		top  int
		want string
	}{
		{7, "Top 10"},
		{25, "Top 25"},
		{26, "Top 50"},
	}

	for _, table := range tables {
		got := topRange(table.top)

		if got != table.want {
			t.Errorf("Waiting for this %s but the return was this: %s", table.want, got)
		}
	}
}

File: JS/TopN.test.js
/**
 * @version JAVASCRIPT
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ node NomeDaClasse.test
 */
const TEST = require('tape')
const OBJ = require('./TopN.js')

TEST('Starting TopN test...', (t) => {
    //Test 1
    t.assert(OBJ.topRange(7) === "Top 10", "Expect: 7 = Top 10")
    //Test 2
    t.assert(OBJ.topRange(25) === "Top 25", "Expect: 25 = Top 25")
    //Test 3
    t.assert(OBJ.topRange(26) === "Top 50", "Expect: 26 = Top 50")

    t.end()
})
